Dockerfile

```
@@ -0,0 +1,15 @@
+FROM node:20 
+
+WORKDIR /app
+
+COPY package.json .
+COPY pnpm-lock.yaml .
+
+RUN npm install -g pnpm
+RUN pnpm install
+
+COPY src ./
+
+EXPOSE 4000 
+
+CMD ["pnpm start:dotenv ", "start:dotenv"]
```

@Srhiulli okay, como isso aqui funcionou não faço ideia kkkkkk

Mas aqui eu faria algo assim:

```suggestion
ENTRYPOINT ["pnpm"]
CMD ["start:dotenv"]
```

Motivos: porque eu acho bunitinho, brincadeira kkkk

O `ENTRYPOINT`, sempre vai ser teu binário padrão, teu comando padrão, então tu sempre vai querer rodar algo tipo: `pnpm <algo>`, então por isso o `pnpm` vai no `ENTRYPOINT`.

Já o `CMD` seria os parámetros que tu quer passar para esse `ENTRYPOINT`, no nosso caso o `start:dotenv`, outra coisa interessante é que o `CMD` pode ser sobrescrito, já o `ENTRYPOINT` não (spoiler: na verdade ele pode sim, tem um jeitinho, mas em via de regra ele não pode)